immer.js

```diff
@@ -66,7 +79,18 @@ function immer(baseState, thunk) {
         if (isPlainObject(base) || Array.isArray(base)) {
             if (isProxy(base)) return base // avoid double wrapping
             if (proxies.has(base)) return proxies.get(base)
-            const proxy = new Proxy(base, objectTraps)
+            let proxyTarget = base
+            // special case, if the base tree is frozen, we cannot modify it's proxy it in strict mode, clone first.
+            if (Object.isFrozen(proxyTarget)) {
+                proxyTarget = Array.isArray(proxyTarget) ? proxyTarget.slice() : { ...proxyTarget }
```

Comment:
Since spread operator is used, `babel-plugin-transform-object-rest-spread` should be added as a dev dependecny. Also, .babelrc should be updated with `"plugins": ["transform-object-rest-spread"]`

Reply:
Good catch, actually I did not put the config there on purpose, to make sure the js files works on modern environments without compilation. I am a bit suprised Jest is fine with this (maybe it is because I switched to node 8). Probably I should add transpilation step soonish...